lib/portlayer/metrics/container.go

```diff
@@ -66,113 +62,103 @@ type MemoryMetrics struct {
+	Kbps    int64  // KiloBytesPerSecond
+	Packets int64  // total packet count
+	Errors  int64  // NOT CURRENTLY IMPLEMENTED
+	Dropped int64  // total dropped packet count
 }
 
-// NetworkMetrics encapsulates available vm Network metrics
-type NetworkMetrics struct {
-	//TBD
+// Network provides metrics for individual network devices
+type Network struct {
+	Name string
+	Rx   NetworkUsage
+	Tx   NetworkUsage
 }
 
-// DiskUsage provides metrics for specific disks
+// DiskUsage provides detailed disk stats
 type DiskUsage struct {
-	//TBD
+	Bytes uint64 // total bytes for interval
+	Kbps  int64  // KiloBytesPerSecond for interval
+	Op    uint64 // Operation count for interval
+	Ops   int64  // Operations per second
 }
 
-// NetworkMetrics encapsulates available vm Storage metrics
-type StorageMetrics struct {
-	//TBD
+// VirtualDisk provides metrics for individual disks
+type VirtualDisk struct {
+	Name  string
+	Write DiskUsage
+	Read  DiskUsage
 }
 
-// VMMetrics encapsulates vm metrics available
+// VMMetrics encapsulates the available metrics
 type VMMetrics struct {
 	CPU        CPUMetrics
 	Memory     MemoryMetrics
-	Network    NetworkMetrics
-	Storage    StorageMetrics
+	Networks   []Network
+	Disks      []VirtualDisk
 	SampleTime time.Time
 	// interval of collection in seconds
 	Interval int32
 }
 
-// collectorVM is the VM metrics collector
-type collectorVM struct {
+// vmCollector is the VM metrics collector
+type vmCollector struct {
 	perfMgr *performance.Manager
+	timer   *time.Ticker
 
 	// subscribers to streaming
 	mu   sync.RWMutex
-	subs map[types.ManagedObjectReference]*subscriber
-}
-
-// subscriber is the receiver of the metrics
-type subscriber struct {
-	id        string
-	ref       types.ManagedObjectReference
-	publisher *pubsub.Publisher
+	subs map[interface{}]*subscription
 }
 
-func newVMCollector(session *session.Session) *collectorVM {
-	s := make(map[types.ManagedObjectReference]*subscriber)
-	cc := &collectorVM{
-		subs:    s,
+// newVMCollector will instantiate a new collector responsible for
+// gathering VM metrics
+func newVMCollector(session *session.Session) *vmCollector {
+	cc := &vmCollector{
+		subs:    make(map[interface{}]*subscription),
 		perfMgr: performance.NewManager(session.Vim25()),
 	}
-
-	// kick off the subscription sampler
-	go cc.sampler()
-
 	return cc
 }
 
-// sampler will check for subscriptions and sample when needed
-func (cc *collectorVM) sampler() {
-	// TODO: replace this with service that only runs when there are
-	// active subscribers
-	d := time.Duration(int64(sampleInterval)) * time.Second
-	for range time.Tick(d) {
-		// if we have no subscribers skip
-		if cc.SubscriberCount() == 0 {
-			continue
+// Start will begin the collection polling process
+func (cc *vmCollector) Start() {
+	// create timer with sampleInterval alignment
+	cc.timer = time.NewTicker(time.Duration(int64(sampleInterval)) * time.Second)
+	// loop on the channel
+	go func() {
+		for {
+			select {
```

Comment:
Yes -- I thought that since Stop() "released the ticker resources" that it would close...but that's not the case.  I've added logic (locally) to break out of the loop -- I'll push up for review in a few.